server/src/main/java/io/spine/server/entity/storage/Columns.java

```diff
@@ -147,94 +107,158 @@ public static void checkColumnDefinitions(EntityClass<?> entityClass) {
+     *         names for storing} to their {@linkplain MemoizedValue memoized values}
+     * @see MemoizedValue
+     */
+    static <E extends Entity<?, ?>> Map<String, MemoizedValue> extractColumnValues(
+            E entity,
+            Collection<EntityColumn> entityColumns) {
+        checkNotNull(entityColumns);
+        checkNotNull(entity);
+
+        final Class<? extends Entity> entityClass = entity.getClass();
+        if (!isPublic(entityClass)) {
+            return Collections.emptyMap();
         }
-        addToCache(entityType);
+
+        final Map<String, MemoizedValue> values = recordColumnValuesToMap(entityColumns, entity);
+
+        return values;
     }
 
     /**
-     * Caches the {@linkplain Entity entity type}
-     * for further {@linkplain EntityColumn columns} retrieving.
+     * Checks if the given {@link Entity entity class} is public.
+     *
+     * <p>Outputs a message to the log if the class is non-public.
+     *
+     * @param entityClass {@link Entity entity class} to check
+     * @return {@code true} if class is public and {@code false} otherwise
      */
-    private static void addToCache(Class<? extends Entity> entityType) {
-        final BeanInfo entityDescriptor;
-        try {
-            entityDescriptor = Introspector.getBeanInfo(entityType);
-        } catch (IntrospectionException e) {
-            throw new IllegalStateException(e);
+    private static boolean isPublic(Class<? extends Entity> entityClass) {
+        checkNotNull(entityClass);
+
+        final int modifiers = entityClass.getModifiers();
+        if (!Modifier.isPublic(modifiers)) {
+            logNonPublicClass(entityClass);
+            return false;
         }
 
-        final Collection<EntityColumn> entityColumns = newLinkedList();
-        for (PropertyDescriptor property : entityDescriptor.getPropertyDescriptors()) {
-            final Method getter = property.getReadMethod();
-            final boolean isEntityColumn = getAnnotatedVersion(getter).isPresent();
-            if (isEntityColumn) {
-                final EntityColumn column = EntityColumn.from(getter);
-                entityColumns.add(column);
-            }
+        return true;
+    }
+
+    /**
+     * Generates {@linkplain MemoizedValue memoized values} for the given {@linkplain EntityColumn columns}
+     * of the given {@link Entity}.
+     *
+     * <p>Records the result to {@link Map}.
+     *
+     * @param columns {@link Collection collection} of columns to extract values from
+     * @param entity  {@link Entity} from which to extract the values
+     * @return a {@link Map} of the column {@linkplain EntityColumn#getStoredName()
+     *         names for storing} to their {@linkplain MemoizedValue memoized values}
+     * @see MemoizedValue
+     */
+    private static <E extends Entity<?, ?>> Map<String, MemoizedValue> recordColumnValuesToMap(
+            Collection<EntityColumn> columns,
```

Comment:
We tend to move the method name to the next line instead of separating the list of arguments, as you did here.